### PR TITLE
Allow changing job settings via restart API

### PR DESCRIPTION
**Pull request by Martchus**
So `openqa-investigate` can use it to handle jobs with parallel
and directly chained dependencies, see
https://progress.opensuse.org/issues/95783.

### CODE REVIEW

File: lib/OpenQA/Schema/Result/Jobs.pm [hunk 1]
```diff
@@ -616,24 +629,24 @@ sub _create_clone {
       && $cluster_job_info->{ok};
 
     # Duplicate settings (with exceptions)
+    my %spec_settings = %$settings;
+    my %main_settings
+      = map { (delete $spec_settings{$_}) // $self->$_ } qw(TEST VERSION ARCH FLAVOR MACHINE BUILD DISTRI);
```
Comment (okurz) [inline lib/OpenQA/Schema/Result/Jobs.pm:634]: We already have a list of such parameters in SCENARIO_KEYS and SCENARIO_WITH_MACHINE_KEYS in lib/OpenQA/Schema/Result/Jobs.pm . I suggest you define another constant in there for what you want to use here. I don't have a good name for that ready though

Comment (Martchus) [reply lib/OpenQA/Schema/Result/Jobs.pm:634]: I've been thinking the same and there's also a very similar list in `create_from_settings`. Maybe I can at least de-duplicate the common denominator.

File: lib/OpenQA/Schema/Result/Jobs.pm [hunk 2]
```diff
@@ -593,6 +593,21 @@ sub is_ok {
     return 0;
 }
 
+sub extract_group_args_from_settings ($settings) {
+
```
Comment (okurz) [inline lib/OpenQA/Schema/Result/Jobs.pm:597]: remove blank line?

File: lib/OpenQA/Schema/Result/Jobs.pm [hunk 3]
```diff
@@ -43,6 +43,9 @@ use OpenQA::WebSockets::Client;
 use constant SCENARIO_KEYS => (qw(DISTRI VERSION FLAVOR ARCH TEST));
 use constant SCENARIO_WITH_MACHINE_KEYS => (SCENARIO_KEYS, 'MACHINE');
 
+# job settings which are defined directly as column of the jobs table
+use constant MAIN_SETTINGS => (qw(DISTRI VERSION FLAVOR ARCH TEST MACHINE BUILD));
```
Comment (kraih) [inline lib/OpenQA/Schema/Result/Jobs.pm:47]: I think the outer `()` does nothing here.

**Review by okurz — APPROVED** [review]
I removed the notice about "still a draft" because obviously it's not anymore :)

**Review by kraih — APPROVED** [review]

**Pull request merged**